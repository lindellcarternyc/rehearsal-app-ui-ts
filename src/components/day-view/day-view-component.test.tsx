import * as React from 'react'
import * as ReactDOM from 'react-dom'
import * as renderer from 'react-test-renderer'

import DayViewComponent, { DayViewComponentProps } from './day-view-component'
const onClick = jest.fn()
const selectRehearsal = jest.fn()
const cancelRehearsal = jest.fn()
const showAddRehearsal = jest.fn()
const showEditRehearsal = jest.fn()

describe('DayViewComponent', () => {
  it('renders without crashing', () => {
    const props: DayViewComponentProps = {
      showAddRehearsal,
      cancelRehearsal,
      showEditRehearsal,
      date: 'Sat, January 20',
      rehearsals: [
        {time: '13:00 - 14:30', material: ['']},
        {time: '14:30 - 16:00', material: ['']},
        {time: '16:00 - 18:00', material: ['']}
      ],
      onClick,
      selectRehearsal
    }
    const div = document.createElement('div')
    ReactDOM.render(
      <DayViewComponent {...props} />,
      div
    )
  })

  it('renders a day with many rehearsals', () => {
    const props: DayViewComponentProps = {
      showAddRehearsal,
      cancelRehearsal,
      showEditRehearsal,
      date: 'Sat, January 20',
      rehearsals: [
        {time: '13:00 - 14:30', material: ['']},
        {time: '14:30 - 16:00', material: ['']},
        {time: '16:00 - 18:00', material: ['']}
      ],
      onClick,
      selectRehearsal
    }

    const tree = renderer.create(
      <DayViewComponent {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a day with one rehearsal', () => {
    const props: DayViewComponentProps = {
      showAddRehearsal,
      cancelRehearsal,
      showEditRehearsal,
      date: 'Tue, January 15',
      rehearsals: [
        {time: '19:00 - 22:00', material: ['']}
      ],
      onClick,
      selectRehearsal
    }
    const tree = renderer.create(
      <DayViewComponent {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})