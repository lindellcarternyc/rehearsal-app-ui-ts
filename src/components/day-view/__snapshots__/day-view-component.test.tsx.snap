// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayViewComponent renders a day with many rehearsals 1`] = `
<div
  className="ui container"
>
  <h2
    className="ui header"
  >
    Sat, January 20
  </h2>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Back to week
  </button>
  <div
    className="ui segment"
  >
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      13:00 - 14:30
    </button>
  </div>
  <div
    className="ui segment"
  >
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      14:30 - 16:00
    </button>
  </div>
  <div
    className="ui segment"
  >
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      16:00 - 18:00
    </button>
  </div>
</div>
`;

exports[`DayViewComponent renders a day with one rehearsal 1`] = `
<div
  className="ui container"
>
  <h2
    className="ui header"
  >
    Tue, January 15
  </h2>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Back to week
  </button>
  <div
    className="ui segment"
  >
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      19:00 - 22:00
    </button>
  </div>
</div>
`;
